### ISSUE TITLE
Argument 4 of mkdir cannot be null

### ISSUE THREAD

**Issue by steal3rd** [vimeo/psalm#10796]
Psalm is reporting the fourth argument of the `mkdir()` function cannot be null, but the function works as expected if used anyway.

The full message is:

> Argument 4 of mkdir cannot be null, null value provided to parameter with type resource

And can be see with this example:

https://psalm.dev/r/6c64902528

```php
mkdir("home/sample", 0755, true); // Works
mkdir("home/sample", 0755, true, null); // Also works
```
It's difficult to find examples of how this parameter would be used, but PHP _does_ support passing a `null` for this argument. The docs describe it that way as well.

https://www.php.net/manual/en/function.mkdir

```php
mkdir(
	string $directory,
	int $permissions = 0777,
	bool $recursive = false,
	?resource $context = null
): bool
```

**Comment by ging-dev** [comment]
> Psalm is reporting the fourth argument of the `mkdir()` function cannot be null, but the function works as expected if used anyway.
> 
> The full message is:
> 
> > Argument 4 of mkdir cannot be null, null value provided to parameter with type resource
> 
> And can be see with this example:
> 
> https://psalm.dev/r/6c64902528
> 
> ```
> mkdir("home/sample", 0755, true); // Works
> mkdir("home/sample", 0755, true, null); // Also works
> ```
> 
> It's difficult to find examples of how this parameter would be used, but PHP _does_ support passing a `null` for this argument. The docs describe it that way as well.
> 
> https://www.php.net/manual/en/function.mkdir
> 
> ```
> mkdir(
> 	string $directory,
> 	int $permissions = 0777,
> 	bool $recursive = false,
> 	?resource $context = null
> ): bool
> ```
You can fix callmap signature in this line: https://github.com/vimeo/psalm/blob/dca888c777aabd07aae4d2dc126cb90c5a432b90/dictionaries/CallMap.php#L6860

**Comment by weirdan** [comment]
Note that PHP versions before 7.3 did not accept `null`: https://3v4l.org/APVEI
